azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -470,45 +472,81 @@ def disable(self, reason, disable_cgroups):
             except Exception as exception:
                 log_cgroup_warning("Error disabling cgroups: {0}".format(ustr(exception)))
 
-        def _set_cpu_quota(self, unit_name, quota):
-            """
-            Sets CPU quota to the given percentage (100% == 1 CPU)
+        def _get_resource_quotas(self):
+            if self._cgroups_api is None:
+                return []
+            return [CpuQuota(self._cgroups_api), MemoryQuota(self._cgroups_api)]
 
-            NOTE: This is done using a systemtcl set-property --runtime; any local overrides in /etc folder on the VM will take precedence
-            over this setting.
+        def _set_resource_quota(self, unit_name, resource_type, quota):
             """
-            if self._cgroups_api.can_enforce_cpu():
-                quota_percentage = "{0}%".format(quota)
-                log_cgroup_info("Setting {0}'s CPUQuota to {1}".format(unit_name, quota_percentage))
-                CGroupConfigurator._Impl._try_set_cpu_quota(unit_name, quota_percentage)
-
-        def _reset_cpu_quota(self, unit_name):
+            Sets the quota for the given resource type ('CPU', 'Memory', or 'All').
             """
-            Removes any CPUQuota on the agent
+            try:
+                quotas = self._get_resource_quotas()
+                property_names = []
+                values = []
+                if resource_type == ResourceName.ALL:
```

Comment:
let's standardize in one of 'type' or 'name'. mixing them makes the code confusing